.github/workflows/snap.yml

```diff
@@ -94,14 +94,19 @@ jobs:
         with:
           snapcraft-args: remote-build --build-for=${{ matrix.arch }} --launchpad-accept-public-upload
 
+      - name: Rename the logs file, for colons are not allowed in the artifact name
+        if: always()
+        run: cat snapcraft-${{ env.SNAP_NAME }}-*.txt > snapcraft-logs.txt || echo "no log files"
+
       - uses: actions/upload-artifact@v4
-        if: failure()
+        if: always()
```

Comment:
Were there cases where the job succeeded but we needed the logs? It seems like this uploads a lot of noise in cases where we don't generally need it. We don't really see the noise and I guess GitHub covers the cost, but it does seem a bit inefficient.

Reply:
I agree -- I wouldn't think we need this in success cases.

Also, isn't `if: always()` equivalent to not having the `if:` line at all?

Reply:
I think `if: always()` includes cancellations and failures, and `if: success()` is the default.